survlime/survlime_explainer.py

```diff
@@ -137,59 +164,72 @@ def explain_instance(
+        type_fn: Literal["survival", "cumulative"],
         num_samples: int,
         weights: np.ndarray,
         H0: np.ndarray,
-        scaled_data: np.ndarray,
+        data: np.ndarray,
         norm: Union[float, str],
         verbose: bool,
     ) -> Tuple[np.ndarray, float]:
         """Solves the convex problem proposed in: https://arxiv.org/pdf/2003.08371.pdfF
         Args:
-            predict_fn (Callable): function to compute the cumulative hazard.
-            num_samples (int): number of neighbours.
-            weights (np.ndarray): distance weights computed for each data point.
-            H0 (np.ndarray): baseline cumulative hazard.
-            scaled_data (np.ndarray): original data point and the computed neighbours.
-            norm (Union[float, str]: number of the norm to be computed in the cvx problem.
-            verbose (bool): activate verbosity of the cvxpy solver.
+            predict_fn (Callable): function to compute the cumulative hazard
+            type_fn (Literal["survival", "cumulative"]): whether predict_fn is the cumulative hazard funtion or survival function
+            num_samples (int): number of neighbours
+            weights (np.ndarray): distance weights computed for each data point
+            H0 (np.ndarray): baseline cumulative hazard
+            data (np.ndarray): original data point and the computed neighbours
+            norm (Union[float, str]: number of the norm to be computed in the cvx problem
+            verbose (bool): activate verbosity of the cvxpy solver
 
         Returns:
-            b.values (np.ndarray): obtained weights from the convex problem.
-            result (float): residual value of the convex problem.
+            b.values (np.ndarray): obtained weights from the convex problem
+            result (float): residual value of the convex problem
         """
-        epsilon = 0.00000001
-        num_features = scaled_data.shape[1]
+        epsilon = 10 ** (-6)
+        num_features = data.shape[1]
         unique_times_to_event = np.sort(np.unique(self.train_times))
         m = unique_times_to_event.shape[0]
-        if self.column_names is not None:
-            scaled_data_format = pd.DataFrame(scaled_data, columns=self.column_names)
-        else:
-            scaled_data_format = np.copy(scaled_data)
-        H_i_j_wc = predict_wrapper(
+        FN_pred = predict_wrapper(
             predict_fn=predict_fn,
-            data=scaled_data_format,
+            data=data,
             unique_times_to_event=unique_times_to_event,
             model_output_times=self.model_output_times,
         )
-        log_correction = np.divide(H_i_j_wc, np.log(H_i_j_wc + epsilon))
+
+        # From now on, original data must be a numpy array
+        if isinstance(data, np.ndarray):
+            data_np = data
+        elif isinstance(data, pd.DataFrame):
+            data_np = data.to_numpy()
+        else:
+            raise NotImplemented("Unknown data type")
```

Comment:
We could change the error message to 'f"Unknown data type {type(data)} only np.ndarray or pd.DataFrame allowed"'.

This way we give some actionable error message to the user.